sections/Claim/ConvertSection/ConvertSection.tsx

```diff
@@ -0,0 +1,296 @@
+	const [showConverterTxModal, setConverterTxModal] = useState<boolean>(false);
+	const [gasvAelinConverterLimitEstimate, setvAelinConverterGasLimitEstimate] =
+		useState<GasLimitEstimate>(null);
+
+	const { gasPrice, setGasPrice, setTxState } = TransactionData.useContainer();
+	const { monitorTransaction } = TransactionNotifier.useContainer();
+	const { walletAddress, network, signer } = Connector.useContainer();
+
+	const vAelinConverterContract = useMemo(() => {
+		if (!signer || !network?.id) return null;
+
+		const vAelinConverterContract = (getKeyValue(VAelinConverterContract) as any)(
+			network?.id ?? DEFAULT_NETWORK_ID
+		);
+
+		if (!vAelinConverterContract) return null;
+
+		return new ethers.Contract(
+			vAelinConverterContract.address,
+			vAelinConverterContract.abi,
+			signer
+		);
+	}, [signer, network?.id]);
+
+	const vAelinTokenContract = useMemo(() => {
+		if (!signer || !network?.id) return null;
+
+		const vAelinTokenContract = (getKeyValue(VAelinTokenContract) as any)(
+			network?.id ?? DEFAULT_NETWORK_ID
+		);
+
+		if (!vAelinTokenContract) return null;
+
+		return new ethers.Contract(vAelinTokenContract.address, vAelinTokenContract.abi, signer);
+	}, [signer, network?.id]);
+
+	useEffect(() => {
+		const getGasLimitEstimate = async () => {
+			if (!walletAddress || !vAelinConverterContract || !vAelinTokenContract) return;
+			try {
+				setvAelinConverterGasLimitEstimate(null);
+				if (!hasAllowance) {
+					const gasLimit = await vAelinTokenContract.estimateGas.approve(
+						vAelinConverterContract.address,
+						ethers.constants.MaxUint256
+					);
+					setvAelinConverterGasLimitEstimate(wei(gasLimit, 0));
+				} else {
+					const gasEstimate = await vAelinConverterContract.estimateGas.convertAll();
+					setvAelinConverterGasLimitEstimate(wei(gasEstimate, 0));
+				}
+			} catch (e) {
+				console.log(e);
+				setvAelinConverterGasLimitEstimate(null);
+			}
+		};
+		getGasLimitEstimate();
+	}, [walletAddress, network.id, vAelinConverterContract, vAelinTokenContract, hasAllowance]);
+
+	const getAllowance = useCallback(async () => {
+		if (!vAelinConverterContract || !vAelinTokenContract || !walletAddress) return;
+		try {
+			const allowance = await vAelinTokenContract.allowance(
+				walletAddress,
+				vAelinConverterContract.address
+			);
+			setHasAllowance(!!Number(ethers.utils.formatEther(allowance)));
+		} catch (e) {
+			console.log(e);
+			setHasAllowance(false);
+		}
+	}, [vAelinTokenContract, vAelinConverterContract, walletAddress]);
+
+	const getVAelinBalance = useCallback(async () => {
+		if (!vAelinTokenContract || !walletAddress) return;
+
+		try {
+			const tx = await vAelinTokenContract.balanceOf(walletAddress);
+			setHasVAelinBalance(!!Number(ethers.utils.formatEther(tx)));
+		} catch (e) {
+			console.log(e);
+			setHasVAelinBalance(false);
+		}
+	}, [vAelinTokenContract, walletAddress]);
+
+	const handleConverAll = useCallback(async () => {
```

Comment:
Typo. Can you install a spell checker in your IDE? It helps a lot trust me.

Reply:
Done